test/LanguageExpressionsTest.cpp

```diff
@@ -0,0 +1,274 @@
+  EXPECT_THAT(result, ::testing::ElementsAreArray(expected));
+};
+
+// ____________________________________________________________________________
+TEST(LanguageTagGetter, testLanguageTagValueGetterWithoutVocabId) {
+  TestContext testContext{};
+  LanguageTagGetter langTagGetter{};
+  Id dateId1 = Id::makeFromDate(
+         DateOrLargeYear::parseXsdDatetime("1900-12-13T03:12:00.33Z")),
+     dateId2 = Id::makeFromDate(DateOrLargeYear::parseXsdDate("2025-01-01")),
+     boolId1 = Id::makeFromBool(false), boolId2 = Id::makeFromBool(true),
+     intId = Id::makeFromInt(323), doubleId = Id::makeFromDouble(234.23);
+  Id maxId = Id::max(), U = Id::makeUndefined();
+
+  // define the input containing IDs (non literal) and corresponding expected
+  // return values w.r.t. LanguageTagValueGetter
+  std::vector<Id> in = {dateId1, dateId2, boolId1, boolId2, intId, doubleId, U};
+  std::vector<strOpt> expected(7, std::nullopt);
+
+  assertLangTagValueGetter(in, expected, langTagGetter, testContext);
+
+  // test corner case (should trigger AD_FAIL())
+  AD_EXPECT_THROW_WITH_MESSAGE(LanguageTagGetter{}(maxId, &testContext.context),
+                               ::testing::HasSubstr("should be unreachable"));
+}
+
+// ____________________________________________________________________________
+TEST(LanguageTagGetter, testLanguageTagValueGetterWithVocab) {
+  TestContext testContext{};
+  LanguageTagGetter langTagGetter{};
+
+  // input and expectation values w.r.t. Vocab-IDs
+  std::vector<Id> in = {testContext.litId1, testContext.litId2,
+                        testContext.litId3, testContext.litId4,
+                        testContext.litId5, testContext.litId6,
+                        testContext.iriId1, testContext.iriId2};
+  std::vector<strOpt> expected = {"",   "es",    "de-LATN-CH", "de-DE",
+                                  "de", "de-AT", std::nullopt, std::nullopt};
+  assertLangTagValueGetter(in, expected, langTagGetter, testContext);
+}
+
+// ____________________________________________________________________________
+TEST(LanguageTagGetter, testLanguageTagValueGetterWithLocalVocab) {
+  TestContext testContext{};
+  LanguageTagGetter langTagGetter{};
+
+  // input and expectation values w.r.t. LocalVocab-IDs
+  std::vector<Id> in = {testContext.locVocIri1, testContext.locVocIri2,
+                        testContext.locVocLit1, testContext.locVocLit2,
+                        testContext.locVocLit3, testContext.locVocLit4};
+  std::vector<strOpt> expected = {std::nullopt, std::nullopt, "",
+                                  "de-DE",      "de",         "de-AT"};
+  assertLangTagValueGetter(in, expected, langTagGetter, testContext);
+}
+
+// ____________________________________________________________________________
+TEST(LangExpression, testLangExpressionOnLiteralColumn) {
+  testLanguageExpressions<getLangExpression, IdOrLiteralOrIri>(
+      {litOrIri(""), litOrIri("es"), litOrIri("de-LATN-CH"), litOrIri("de-DE"),
```

Comment:
one question:
Is `LANGMATCHES` case insensitive, and if yes, are you dealing with this (I think a simple ascii tolower should work, as language tags only should be ascii).

Reply:
`Langmatches` should be case sensitive given that the tags should conform to the `BPC47` standard, and the valid `language-range` is defined as: `language-range   = (1*8ALPHA *("-" 1*8alphanum)) / "*"` with `alphanum  = ALPHA / DIGIT`. Should i check this with a regex like: `^(?:[A-Z]{1,8}(?:-[a-zA-Z0-9]{1,8})*|*)$`?

Reply:
`Langmatches()` should be case sensitive if the `language-range` complies with the definition: `language-range  = (1*8ALPHA *("-" 1*8alphanum)) / "*"` and `alphanum = ALPHA / DIGIT` given by the `Basic-Language-Range` definition in
https://www.ietf.org/rfc/rfc4647.txt. Should i check this with the following regex: `^(?:[A-Z]{1,8}(?:-[a-zA-Z0-9]{1,8})*|*)$`?

Reply:
Yes it should be case insensitive (solved with tolower now). Should in addition be checked if a valid `language-range` is given?